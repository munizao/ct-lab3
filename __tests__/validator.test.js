const {
  Validator
} = require('../lib/validator.js');

describe ('validator class module', () => {
  const nameValidator = new Validator('name', {
    type: String,
    required: true
  });

  const foodsValidator = new Validator('foods', {
    type: Array,
    required: true
  });

  const weightValidator = new Validator('weight', {
    type: Number,
    required: false
  });


  const goodDog = {
    name: 'spot',
    foods: ['corn', 'spinach'],
    weight: 30
  };
  
  const badDog = {
    name: 'arlo',
    foods: 30,
    weight: [1, 2, 3, 4]
  };

  const emptyDog = {
  };

  it('properly makes a validator', () => {
    expect(nameValidator.field).toEqual('name');
    expect(nameValidator.configuration).toEqual({
      type: String,
      required: true
    });
    expect(nameValidator.validate).toBeDefined();
  });

  it('properly validates', () => {
    expect(nameValidator.validate(goodDog)).toEqual('spot');
    expect(() => foodsValidator.validate(badDog)).toThrowErrorMatchingSnapshot();
    expect(() => nameValidator.validate(emptyDog)).toThrowErrorMatchingSnapshot();
    expect(weightValidator.validate(goodDog)).toEqual(30);
    expect(() => weightValidator.validate(badDog)).toThrowErrorMatchingSnapshot();
    expect(weightValidator.validate(emptyDog)).toEqual(undefined);
  });
});
