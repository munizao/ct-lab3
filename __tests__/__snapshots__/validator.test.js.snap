// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validator class module properly validates 1`] = `"Cannot cast >>30<< to Array"`;

exports[`validator class module properly validates 2`] = `"required field is missing"`;

exports[`validator class module properly validates 3`] = `"Cannot cast >>1,2,3,4<< to Number"`;
